ext/nnstreamer/tensor_filter/tensor_filter_llamacpp.cc

```diff
@@ -0,0 +1,339 @@
+/**
+ * GStreamer Tensor_Filter, llamacpp Module
+ * Copyright (C) 2024 Yelin Jeong <yelini.jeong@samsung.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * @file    tensor_filter_llamacpp.cc
+ * @date    5 Dec 2024
+ * @brief   llamacpp module for tensor_filter gstreamer plugin
+ * @see     http://github.com/nnstreamer/nnstreamer
+ *          https://github.com/ggerganov/llama.cpp
+ * @author  Yelin Jeong <yelini.jeong@samsung.com>
+ * @bug     No known bugs except for NYI items
+ * @details Pipeline example
+ *          gst-launch-1.0 filesrc location=input.txt ! application/octet-stream !
+ *          tensor_converter ! other/tensors,format=flexible !
+ *          tensor_filter framework=llamacpp model=llama-2-7b-chat.Q2_K.gguf invoke-dynamic=TRUE !
+ *          other/tensors,format=flexible ! tensor_decoder mode=octet_stream !
+ *          application/octet-stream ! filesink location=output.txt
+ */
+
+#include <assert.h>
+#include <errno.h>
+#include <nnstreamer_cppplugin_api_filter.hh>
+#include <nnstreamer_log.h>
+#include <nnstreamer_plugin_api_util.h>
+#include <nnstreamer_util.h>
+#include <vector>
+
+#include "ggml.h"
+#include "llama.h"
+/**
+ * @brief Macro for debug mode.
+ */
+#ifndef DBG
+#define DBG FALSE
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif /* __cplusplus */
+void init_filter_llama (void) __attribute__ ((constructor));
+void fini_filter_llama (void) __attribute__ ((destructor));
+#ifdef __cplusplus
+}
+#endif /* __cplusplus */
+
+namespace nnstreamer
+{
+namespace tensorfilter_llamacpp
+{
+/**
+ * @brief Class for TensorFilterLlamaCpp subplugin
+ */
+class TensorFilterLlamaCpp : public tensor_filter_subplugin
+{
+  public:
+  TensorFilterLlamaCpp ();
+  ~TensorFilterLlamaCpp ();
+
+  /* mandatory methods */
+  tensor_filter_subplugin &getEmptyInstance ();
+  void configure_instance (const GstTensorFilterProperties *prop);
+  void invoke (const GstTensorMemory *input, GstTensorMemory *output);
+  void invoke_dynamic (GstTensorFilterProperties *prop,
+      const GstTensorMemory *input, GstTensorMemory *output);
+  void getFrameworkInfo (GstTensorFilterFrameworkInfo &info);
+  int getModelInfo (model_info_ops ops, GstTensorsInfo &in_info, GstTensorsInfo &out_info);
+  int eventHandler (event_ops ops, GstTensorFilterFrameworkEventData &data);
+
+  /* static methods */
+  static void init_filter_llama ();
+  static void fini_filter_llama ();
+
+  private:
+  static TensorFilterLlamaCpp *registered;
+  static const char *name;
+  static const accl_hw hw_list[];
+  static const int num_hw;
+
+  llama_context *ctx;
+  llama_model *model = nullptr;
+  llama_sampler *smpl = nullptr;
+  llama_model_params model_params;
+  llama_sampler_chain_params sampler_params;
+  llama_context_params ctx_params;
+  llama_batch batch;
```

Comment:
batch is used only in ```TensorFilterLlamaCpp::invoke_dynamic``` and its value is never reused.

Reply:
Removed `batch` from class member value and moved it to `invoke_dynamic`.
Thank you!